### PR TITLE
Get/api/articles task 4

### PR DESCRIPTION
**Pull request by crypticalfish86**
Articles complete

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -49,4 +48,54 @@ describe('APP be-nc-news', () =>
...
+                    expect(element.body).toEqual(undefined)
+                })
+            })
+        })
+        test('has sorted the objects in the array by the created_at property, the first date should have an article id of 7', () =>
+        {
+            return request(app).get('/api/articles').expect(200)
+            .then(({body}) =>
+            {
+                expect(body[0].article_id).toEqual(7)
```
Comment (xViolaine) [inline __tests__/app.test.js]: I would argue that this is probably not enough of a test to check that they are sorted in a descending order - have a look at jest sorted

File: app/app.js [hunk 2]
```diff
@@ -1,17 +1,23 @@
 const express = require('express')
-const {getTopics} = require('./controller.js')
+const {getTopics, getArticles} = require('./controller.js')
 const app = express()
 app.use(express.json())
```
Comment (xViolaine) [inline app/app.js]: We don't use app.use(express.json()) for anything yet, so it shouldn't be in your code yet!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -49,4 +48,54 @@ describe('APP be-nc-news', () =>
             })
         })
     })
+    describe('GET /api/articles', () =>
+    {
+        test('GET: will return a status of 200', () => 
+        {
+            return request(app).get('/api/articles').expect(200)
```
Comment (xViolaine) [inline __tests__/app.test.js]: its kind of redundant to have a test for this when you are also testing for this in the next tests :)

File: app/app.js [hunk 4]
```diff
@@ -1,17 +1,23 @@
...
-app.use((error, request, response, next) =>
+app.use((err, request, response, next) => 
 {
-    if(error)
+    console.log(err)
+    response.status(err.status).send(err.msg)
+})
```
Comment (xViolaine) [inline app/app.js]: This doesnt seem to be doing anything - it doesn't have any conditional logic and thus will hit on any request

File: app/app.js [hunk 5]
```diff
@@ -1,17 +1,23 @@
...
+app.use((err, request, response, next) =>
+{
+    if(err)
     {
-        response.status(500).send({error})
+        response.status(500).send({err})
     }
 })
```
Comment (xViolaine) [inline app/app.js]: Generally, all error handlers should be at the bottom of the app.js

File: app/model.js [hunk 6]
```diff
@@ -11,4 +11,33 @@ fetchTopics = () =>
...
+{
+    return db.query
+    (
+        `
+        SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at,
+        articles.votes, articles.article_img_url,
+        (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id)
+        AS comment_count
+        FROM articles
+        JOIN comments
```
Comment (xViolaine) [inline app/model.js]: Is a Join sufficient here?

**Review by xViolaine — CHANGES_REQUESTED** [review]

File: app/model.js [hunk 7]
```diff
@@ -21,7 +21,7 @@ const fetchArticles = () =>
         (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id)
         AS comment_count
         FROM articles
-        JOIN comments
+        RIGHT JOIN comments
```
Comment (Hard-Luck) [inline app/model.js]: Right join would only join the articles with comments, we want to join all articles even if they don't have comments, use LEFT JOIN

**Pull request merged**